scheduler/src/cook/mesos/util.clj

```diff
@@ -338,20 +346,67 @@
 (def ^:const job-states #{"running" "waiting" "completed"})
 (def ^:const instance-states #{"success" "failed"})
 
+;; get-jobs-by-user-and-state-and-submit-by-state is a bit opaque
+;; because it is reaching into datomic internals. Here is a quick explanation.
+;; seek-datoms provides a pointer into the raw datomic indices
+;; that we can then seek through. We set the pointer to look
+;; through the avet index, with attribute :job/state, seek to
+;; state and then seek to the entity id that *would* have been
+;; created at expanded start. This works because the submission
+;; time (which we sort on) is correlated with the entitiy id.
+;; We then seek through the list of all jobs in that state,
+;; then filtering to the target user.
+;;
+;; This differs from get-active-jobs-by-user-and-state as it
+;; is also looking up based on task state.
```

Comment:
It seems to say this "differs" from `get-active-jobs-by-user-and-state` given that `get-active-jobs-by-user-and-state` calls this function

Reply:
Good catch. Merge/rebase bug.